lucet-module/src/version_info.rs

```diff
@@ -0,0 +1,112 @@
+/// VersionInfo is information about a Lucet module to allow the Lucet runtime to determine if or
+/// how the module can be loaded, if so requested. The information here describes implementation
+/// details in runtime support for `lucetc`-produced modules, and nothing higher level.
+#[repr(C)]
+#[derive(Debug, Clone)]
+pub struct VersionInfo {
+    major: u8,
+    minor: u8,
+    patch: u8,
+    reserved: u8,
+    pub capabilities: Capabilities,
+}
+
+impl VersionInfo {
+    pub fn current() -> Self {
```

Comment:
My expectation is that `lucet-runtime`'s version check would always use `VersionInfo::current`, sets supported capabilities, and then does a comparison with whatever we loaded. That comparison function is conspicuously absent here; I think we tend to know what will be breaking change to a module's ABI, and can just bump these fields accordingly, but that's where I'd really like thoughts